c++/src/kj/test.c++

```diff
@@ -346,6 +346,8 @@ public:
     if (passCount > 0) write(GREEN, kj::str(passCount, " test(s) passed"), "");
     if (failCount > 0) write(RED, kj::str(failCount, " test(s) failed"), "");
     context.exit();
+
+    return true;
```

Comment:
Yes, I care that the build succeeds with the warning settings fond at the top of `super-test.sh`, namely:

    -Wall -Wextra -Wno-strict-aliasing -Wno-sign-compare -Wno-unused-parameter

However, for GCC (not Clang), we also add `-Wno-maybe-uninitialized` because it seems to report a lot of bogus warnings of this type, especially related to kj::Maybe. The test script uses Valgrind to check for actual uninitialized variable usage, so the GCC warning is unnecessary.